src/app/services/server.service.ts

```diff
@@ -487,4 +490,17 @@ export class ServerService {
       this.sendError(res, err, false, 500);
     });
   }
+
+  /**
+   * Make sure environment is active
+   *
+   * @param uuid - uuid of environment
+   */
+  private setActiveEnvironment (uuid) {
+    if (this.store.getActiveEnvironment().uuid !== uuid) {
+      this.store.update(
+        setActiveEnvironmentAction(uuid)
+      );
+    }
+  }
```

Comment:
I added this, when I noticed that only the active environment showed the indicator that it is running, after looping through all envs in `START_ALL_ENVIRONMENTS`. The environments where running, but in the UI it still looked like the service needs to be started. As far as I can see, this is due to how the `ActionTypes.UPDATE_ENVIRONMENT_STATUS` reducer is updating based on the `activeEnvironmentUUID`, not the one that is actually changed.

Reply:
You are right, I forgot about this. 
But then I would rather modify the action and the reducer to pass an environmentUUID, instead of artificially switching the active environment, which makes the user experience worse.

Reply:
Good point. I was not sure first how often this action is actually used, but since those are the only two occurences it's easier to update it directly.

Reply:
Thank you for the modification. 
But then I think you can remove the setActiveEnvironment method you created in server.service.ts :)

Reply:
Ah dang, did I forget to remove it? Will do it right away